### DISCUSSION TITLE
Issues with workloads

### DISCUSSION THREAD

**Discussion by jeromelaban** [unoplatform/uno#12668] [Q&A] [answered]
[_From Discord_:](https://discord.com/channels/372137812037730304/550416151172087808/1121129641780985986)

Hey Folks - I've started getting a strange problem with I update uno-check, now when I run it from an administrator shell I get the following screenshot. It used to complete clean. 

I've added a bunch of software onto my machine for example preview Visual Studio 2022 but I can't say which app might have been responsible (if that was the cause),  is there a logfile which might shed more light.

BTW - I need my machine constantly telling me about a pending reboot which I fixed and the reboot warning went in uno-check but this issue was still the same.

![image](https://github.com/unoplatform/uno/assets/5839577/cc5b365e-cf33-4902-96f3-6152c03764b3)

**Comment by jeromelaban** [comment]
This looks like it may be related to the previews being installed. Could you provide the full log of uno-check? There are additional information above that are not visible here.

**Reply by johnkattenhorn** [reply]
I re-ran `uno-check` as `uno-check -n -f -v > uno-check.log`, and I've attached the file it produced.
[uno-check.log](https://github.com/unoplatform/uno/files/11821895/uno-check.log)

I can't think of anything else right now but I'm starting not to trust the installation of Windows on this machine, I've got as far as I remember the same configuration and versions of things on my laptop (because I do the config using a `chocolatey` script and I don't have this issue.

**Reply by jeromelaban** [reply]
Thanks. It looks like a mix of versions, and since you have .NET 8 installed, you may want to try using [`--pre-major`](https://platform.uno/docs/articles/external/uno.check/doc/using-uno-check.html?q=uno-check#install-and-run-uno-check) to get the right workloads installed.

On a side note, we know that this particular workflow is difficult, and we hope that .NET 8 will simplify all of this more.

**Reply by johnkattenhorn** [reply]
Ok I ran that command, and it download newer versions as predicted but still failed, I've attached a new log
[uno-check-pre-major.log](https://github.com/unoplatform/uno/files/11822195/uno-check-pre-major.log)

**Reply by jeromelaban** [reply]
Thanks. It would seem that the workloads installation is corrupted somehow.

In a folder that is not a .NET project, run `dotnet --version`, ensure that you have .NET 8 selected, then run the following:

```
dotnet workload clean --all
```

This will uninstall all workloads for all dotnet versions, and you may be able to re-run uno-check with `--pre-major` to get workloads installed again.

**Comment by johnkattenhorn** [comment]
Here is the output from the `dotnet workload clean --all` the dotnet --version returned `8.0.100-preview.5.23303.2`

<img width="866" alt="Screenshot 2023-06-21 203249" src="https://github.com/unoplatform/uno/assets/616249/6dd3d8ed-2303-4ee9-85ee-8d693095075e">

I then proceeded to do a `uno-check --premajor -f -n` and it was the same or similar error.
<img width="867" alt="Screenshot 2023-06-21 203427" src="https://github.com/unoplatform/uno/assets/616249/5b54ea0a-ec8d-40db-a077-7efe170c1389">

**Reply by jeromelaban** [reply]
Thanks. If you remove the --pre-major flag, is the result similar?

**Reply by johnkattenhorn** [reply]
Same issue - different version numbers 

<img width="875" alt="image" src="https://github.com/unoplatform/uno/assets/616249/07a3d5fc-56db-45ee-a12d-8b4dcf76a39d">

**Reply by jeromelaban** [reply]
Thanks. 

I've published a new uno-check preview that updates both net7 and net8 targets that use recent workloads. You can install it using `dotnet tool install uno.check --version 1.13.0-dev.23`. You may need to run `dotnet tool uninstall -g uno.check first`.

Can you try again with and without `--pre-major` flag and paste the results here?

**Reply by johnkattenhorn** [reply]
Yep - of course - thanks I'll try it now

**Reply by johnkattenhorn** [reply]
OK - it did'nt work (I like the new template check though), here is the logfile
[uno-check.log](https://github.com/unoplatform/uno/files/11833887/uno-check.log)

**Reply by johnkattenhorn** [reply]
Sorry that file was blank here is the populated one 
[uno-check.log](https://github.com/unoplatform/uno/files/11833900/uno-check.log)

**Reply by jeromelaban** [reply]
Ah, right, the default manifests are not updated, I'll publish another preview. Can you try again with:

```
uno-check --manifest https://raw.githubusercontent.com/unoplatform/uno.check/9db10686e7853e9db53e5bb2d6221ad978b622d1/manifests/uno.ui.manifest.json
```

Apologies for the multiple tryouts :)

**Reply by johnkattenhorn** [reply]
No worries - thanks for helping ;-) - I'll try it now

**Reply by johnkattenhorn** [reply]
It's downloaded something but still failed the re-check.

[uno-check.log](https://github.com/unoplatform/uno/files/11834430/uno-check.log)

**Reply by jeromelaban** [reply]
Thanks. As it still fails, it would mean that the .NET installation is broken somehow. Still, are you able to build apps properly?

**Reply by johnkattenhorn** [reply, marked as answer]
Hey @jeromelaban, 

I've now resolved this issue - you were correct - something was wrong with the Visual Studio 2022 install.

Step to Resolve

1) Uninstalled Visual Studio 2022
2) Uninstalled Visual Studio 2022 Preview
3) Checked Add/Remove Programs and Removed the Visual Studio Installer
4) Ran the Visual Studio CleanUp Exe list in this article (https://learn.microsoft.com/en-us/visualstudio/install/uninstall-visual-studio?view=vs-2019
5) Rebooted
6) Re-Installed Visual Studio 2022 with the recommended workloads for Uno (and some which I need for other projects)
7) Ran uno-check with the command-line you gave me originally
8) Same as before, it detected missing workloads and installed them, but this time didn't fail the second check
9) I ran it again with the verbose switch and captured the log for your information which I've attached.

So it definitely looks like something broke my install probably last week, but I can't recall what it was; the only thing I remember was going through a Visual Studio update from VS Installer for both Preview and Enterprise.

[uno-check.log](https://github.com/unoplatform/uno/files/11845743/uno-check.log)

**Reply by jeromelaban** [reply]
Thanks for the update! We really with that .NET 8 will help to resolve this kind of situations. We'll still update uno-check to align with the latest stable builds.

**Reply by limeniye** [reply]
I did this steps again and removed all installed .Net SKDs (7/8-preview).
It helps me. Thanks.
